### PR TITLE
Move xref related code to a new file, readtags-xref.el

### PR DESCRIPTION
**Pull request by masatake**
This code separation may help @rrthomas study the internal of citre.

### CODE REVIEW

**Comment by masatake** [comment]
I have two questions.

* How do you think having global-citre-mode?
  When testing readtags-xref backend, I have to eval ` (add-hook 'xref-backend-functions #'citre-xref-backend nil t)` evenry time after jumping.

* How do you think that you maintain this repo at Universal-ctags organization so that more people can find citre when your "busy" marker can be removed.

**Comment by AmaiKinono** [comment]
> How do you think having `global-citre-mode`?

I could add it as long as someone needs it. Personally I think it's not that useful. For example, I definitely won't use `citre-mode` when editing a markdown file. I would recommend add `citre-mode` to `prog-mode-hook` or any major mode hook you want.

Users could also implement their own function to see if current file needs `citre-mode` (e.g. based on some variable, and set it in `.dir-locals.el` in projects that need `citre-mode`), and add that to `find-file-hook`.

>  I have to eval `(add-hook 'xref-backend-functions #'citre-xref-backend nil t)` every time after jumping.

I think doing `M-x citre-mode` after jumping should work since it does what you did. Or you could setup the hook like above.

> How do you think that you maintain this repo at Universal-ctags organization

A wonderful idea ;) To be honest, I'm still not very sure if I could finish this project. If I could actually release 0.1, I'll ask you then.

> ...when your "busy" marker can be removed

Well, just to tell you, that actually I'm writing my master's thesis. The main work are already done, so I can work on OSS from time to time at this period. I expect to remove the "busy" marker at the end of this year.

**Comment by AmaiKinono** [comment]
I'm good with the changes, but not sure on the filename. It's inconsistant with `citre-readtags`. We can:

- Change it (and also the feature it provides) to `citre-xref`.
- Or keep it, and change `citre-readtags`, including all symbol prefixes in it, to `readtags`.

I don't know if the common practice is to prefix every elisp library in a project with the same name. From what I've seen it seems so.

In either way the symbols in `readtags-xref.el` should be prefixed by the name of the feature.

File: readtags-xref.el [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        (line-content (citre-readtags-get-field 'line-content record)))
+    (xref-make
+     (concat
+      (propertize kind 'face 'warning) " " line-content)
+     (xref-make-file-location path line 0))))
+
+(defun citre--xref-find-definition (symbol)
+  "Return the xref object of the definition information of SYMBOL."
+  (mapcar #'citre--make-xref-object
+          (citre-get-definition-records nil symbol)))
```
Comment (AmaiKinono) [inline readtags-xref.el:54]: This is (the only) function from `citre.el`, so this file needs a `(require 'citre)`

Comment (masatake) [reply readtags-xref.el:54]: I didn't recognize this.

**Comment by masatake** [comment]
I would like to withdraw this. Sorry for making noise.
I would like to retry this topic in the future.

**Comment by AmaiKinono** [comment]
Ok, no problem ;)

**Comment by masatake** [comment]
>>   How do you think that you maintain this repo at Universal-ctags organization

> A wonderful idea ;) To be honest, I'm still not very sure if I could finish this project. If I could actually release 0.1, I'll ask you then.

I sent an invitation. I assume you can transfer the repository if you are a member of the Universal-ctags organization. The benefit is you can transfer an issue between repositories in an organization.
Our interests cross over the repsotires. So trasfering helps us managing issues.
Transfer the ctire repo if you want and when you are ready.

Even if you don't transfer, I would like to say that you should be a member beacuse you know well about tags file format and the concept behind ctags.

**Comment by AmaiKinono** [comment]
Thank you ;) I accepted it.

I would like to transfer, but still, I want to do that after we release 0.1 (or when we are close to it).

**Pull request closed without merging**